### PR TITLE
docs: reorder build and link steps

### PR DESCRIPTION
**Pull request by adammoody**
This changes the docs to list steps in the following order:
- build unifyfs
- integrate the unifyfs api into an app
- link an app with libunifyfs
- configure
- run

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Testing (addition of new tests or update to current tests)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the UnifyFS code style requirements.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commit messages are properly formatted.

### CODE REVIEW

**Review by CamStan — CHANGES_REQUESTED** [review]
Changing a file name will break any cross-doc links that go to the top of a different file. Links to specific labels within a file should still be fine.
Just need to change the doc name in these links to the new doc name.
E.g.,
```
You can build and run UnifyFS by following :doc:`these instructions <build-intercept>`.
```
becomes
```
You can build and run UnifyFS by following :doc:`these instructions <build>`.
```

Hard to link to exactly where these lines are in GitHub since they get formatted and no line numbers are provided. 
I'll give you a link to which file it is in the repo, and then provide a line number that they're on in an editor.

**Comment by CamStan** [comment]
Would you be willing to change the version number in the README to 0.9.1 in this PR as well?

File: docs/index.rst [hunk 1]
```diff
@@ -14,10 +14,11 @@ UnifyFS: A file system for burst buffers
...
    definitions
    assumptions
-   build-intercept
-   api-mount
+   build
+   api
+   link
    configuration
-   start-stop
+   run
```
[the remark this answers is not in the document]
Comment (adammoody) [reply docs/index.rst:21]: Dang, totally forgot to check for cross links to those files.  Nice catch!  Yeah, I'll clean those up.

**Review by CamStan — APPROVED** [review]
@adammoody thanks for doing this!

**Comment by adammoody** [comment]
Made some more edits after reading through the current pages.  These are probably not perfect, but at some point it will be easier to merge so we can read and test links in the formatted pages on readthedocs.

**Pull request merged**